### PR TITLE
AMQP-119: Add cluster supportability for connector.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java [hunk 1]
```diff
@@ -37,18 +43,45 @@
...
+    public void testSubreceiversAreNotStartedWhenNotPrimaryNodeAndNotOnlyListenOnPrimaryNode() throws Exception
+    {
+        testSubreceiversAreNotStarted(false, false);
+    }
+
+
+    private void testSubreceiversAreNotStarted(boolean primaryPollingInstance, boolean listenOnPrimaryNodeOnly) throws MuleException, CreateException, Exception
+    {
+        setupConnector(primaryPollingInstance, listenOnPrimaryNodeOnly);
+        sleep(TIMEOUT);
```
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java]: Code Smell Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVzotIXnmsUbKcD&open=AWiBQVVzotIXnmsUbKcD)

[ISSUE_KEY]: AWiBQVVzotIXnmsUbKcD

File: mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiver.java [hunk 2]
```diff
@@ -77,6 +81,48 @@ protected synchronized void doConnect() throws Exception
...
 
+        final MuleContext muleContext = amqpConnector.getMuleContext();
+
+        if (muleContext.isPrimaryPollingInstance() || !amqpConnector.isListenOnPrimaryNodeOnly())
+        {
+            generateSubreceivers();
+        }
+        else
+        {
+            muleContext.registerListener(new ClusterNodeNotificationListener<ClusterNodeNotification>()
```
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiver.java]: Code Smell Code Smell: Make this anonymous inner class a lambda ([squid:S1604](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1604))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVQotIXnmsUbKcB&open=AWiBQVVQotIXnmsUbKcB)

[ISSUE_KEY]: AWiBQVVQotIXnmsUbKcB

File: mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java [hunk 3]
```diff
@@ -37,18 +43,45 @@
...
+        verify(mockSubreceiver, never());
```
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java]: Code Smell Code Smell: Complete the assertion. ([squid:S2970](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2970))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVzotIXnmsUbKcC&open=AWiBQVVzotIXnmsUbKcC)

[ISSUE_KEY]: AWiBQVVzotIXnmsUbKcC

File: mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java (hunk 1 above)
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java]: Code Smell Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVzotIXnmsUbKcD&open=AWiBQVVzotIXnmsUbKcD)

[ISSUE_KEY]: AWiBQVVzotIXnmsUbKcD

File: mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java (hunk 3 above)
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/test/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiverTestCase.java]: Code Smell Code Smell: Complete the assertion. ([squid:S2970](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2970))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVzotIXnmsUbKcC&open=AWiBQVVzotIXnmsUbKcC)

[ISSUE_KEY]: AWiBQVVzotIXnmsUbKcC

File: mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiver.java (hunk 2 above)
Comment (sonar-mulesoft) [inline mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/endpoint/receiver/MultiChannelMessageReceiver.java]: Code Smell Code Smell: Make this anonymous inner class a lambda ([squid:S1604](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1604))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-transport-amqp&pullRequest=173&issues=AWiBQVVQotIXnmsUbKcB&open=AWiBQVVQotIXnmsUbKcB)

[ISSUE_KEY]: AWiBQVVQotIXnmsUbKcB

**Comment by fsgonz** [comment]
I'll reopen this PR.

**Pull request closed without merging**